sklearn/model_selection/tests/test_split.py

```diff
@@ -1259,6 +1271,70 @@ def test_train_test_split_default_test_size(train_size, exp_train, exp_test):
     assert len(X_test) == exp_test
 
 
+@pytest.mark.parametrize(
+    "array_namepsace, device, dtype", yield_namespace_device_dtype_combinations()
+)
+@pytest.mark.parametrize(
+    "shuffle,stratify",
+    (
+        (True, None),
+        (True, np.hstack((np.ones(6), np.zeros(4)))),
+        # stratification only works with shuffling
```

Comment:
I think this limitation needs to be a part of the changelog.

Reply:
This isn't a new constraint, `train_test_split` already doesn't support this combination. I added the comment for people from the future who might wonder why we aren't testing all four combinations/think we forgot to add that test case.